starlette/responses.py

```diff
@@ -258,7 +259,7 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
             except OSError:
                 raise ClientDisconnect()
         else:
-            async with anyio.create_task_group() as task_group:
+            async with create_collapsing_task_group() as task_group:
```

Comment:
Do we also have an issue with `StreamingResponse` standalone?

With my comment on the other PR I was trying to avoid the changes here. 🤔

Reply:
Yeah it's the same sort of issue, but it's wrapped in two TaskGroups before the user gets the exception. Collapsing is ok here because the wait_for_disconnect exception is always 'catastrophic' eg won't be caught